### ISSUE TITLE
Add data categories; tag-like structure for data

### ISSUE THREAD

**Issue by yunjae2** [yunjae2/tagit#12]
# Introducing data categories and parsing rules
After recording a command, users may want to parse the result of the command and save refined data into different places.

Example case:
```
$ tagit record myexp "nr_cpu=8, mem=128GB, storage=nvme" -- ./run_exp.sh
```
When output of the command above looks like:
```
...
blah blah blah
throughput: 10GB/s
latency: 20ms
mem_usage: 75GB
foo bar baz
...
```
Users may want to save each metric into different places accessible in different names (such as its metric name).
```
$ tagit report myexp "nr_cpu=8, mem=128GB, storage=nvme" -d "latency"
[myexp] (nr_cpu=8, mem=128GB, storage=nvme)
- latency: 20ms

$ tagit report myexp "nr_cpu=8, mem=128GB, storage=nvme" -d *
[myexp] (nr_cpu=8, mem=128GB, storage=nvme)
- throughput: 10GB/s
- latency: 20ms
- mem_usage: 75GB
```

To support these cases, two new ideas are developed:
1. Data categories; tag-like feature that enables users save different type of data separately
2. Parsing rules; parse the result of the command and save useful data into different data categories

## User interface
```
$ tagit parse add [-s <source category>] <data category> <parsing command>
```
After registering parsing rules, newly added data is automatically parsed as well as previous data.

### Example
```
$ tagit parse add myexp "latency" "awk /^latency/{print $NF}"
$ tagit parse add myexp -s "latency" "latency_number" "sed 's/ms//'"

$ tagit report myexp "nr_cpu=8, mem=128GB, storage=nvme" -d "latency_number"
[myexp] (nr_cpu=8, mem=128GB, storage=nvme)
- latency_number: 20
```

**Comment by yunjae2** [comment]
The default data category is `raw`. That is, the command output is recorded to `raw`.

**Comment by yunjae2** [comment]
Specify the scope of the parsing rule in the parsing command?
Is it really needed?

**Comment by yunjae2** [comment]
We must avoid cyclic parsing rules; if a user tries to add a parsing rule that leads to create a cycle in parsing graph, abort it.

**Comment by yunjae2** [comment]
Also, we must prohibit the two actions:
- Recording directly to the data category that is produced by parsing rules
- Adding parsing rules that writes the parsing output to the data category that is directly recorded

**Comment by yunjae2** [comment]
source data is provided as stdin to parsers.

**Comment by yunjae2** [comment]
This feature will be included in v0.0.5.
